accel/tcg/cputlb.c

```diff
@@ -1426,7 +1445,15 @@ static bool victim_tlb_hit(CPUArchState *env, size_t mmu_idx, size_t index,
         cmp = qatomic_read((target_ulong *)((uintptr_t)vtlb + elt_ofs));
 #endif
 
-        if (cmp == page) {
+        bool tag_write_invalid = false;
+
+#ifdef TARGET_CHERI
+        if (cap_write && (env_tlb(env)->d[mmu_idx].viotlb[vidx].tagmem_write ==
+                          TLBENTRYCAP_INVALID_WRITE))
+            tag_write_invalid = true;
```

Comment:
Do you need this variable, can't we just `return false;` here?

Reply:
We could continue, but not return. tag_write_invalid is true if the current entry in the victim TLB could not be used for tag writes, but it might not actually match the address being looked up. Subsequent entries may be both the page we are looking for and be valid for tag writing.

Reply:
Oh yeah I missed that it's inside the loop. Maybe `continue` instead of adding the extra variable?